### PR TITLE
[11561] Fixing static linking when SECURITY=OFF but SSL is needed

### PR DESCRIPTION
**Pull request by MiguelBarro**
The windows installer binaries require OpenSSL dependency despite of not been build with security.
Now both static and dynamic OpenSSL libraries are distributed. If the user sets *FASTDDS_STATIC=ON* the deployed static libs will be used avoiding any abi break issues.

The **dll hell** on OpenSSL dlls (an independent OpenSSL installation broke Fast-DDS operation) is now solved using a private assembly.

Generated installer available [here](http://jenkins.eprosima.com:8080/job/FastRTPS%20Installer%20Windows/243/)

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -456,13 +457,55 @@ elseif(NOT EPROSIMA_INSTALLER)
     target_link_libraries(${PROJECT_NAME} ${PRIVACY} fastcdr foonathan_memory
         ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS}
         ${TINYXML2_LIBRARY}
-        $<$<BOOL:${LINK_SSL}>:OpenSSL::SSL$<SEMICOLON>OpenSSL::Crypto>
+        $<$<BOOL:${LINK_SSL}>:OpenSSL::SSL$<SEMICOLON>OpenSSL::Crypto$<$<BOOL:${WIN32}>:$<SEMICOLON>crypt32.lib>>
```
Comment (richiware) [inline src/cpp/CMakeLists.txt:460]: I think `$<$<BOOL:${WIN32}>:$<SEMICOLON>crypt32.lib>` is not necessary.

File: cmake/packaging/Config.cmake.in [hunk 2]
```diff
@@ -26,15 +26,16 @@ find_package(fastcdr REQUIRED)
...
 @FASTRTPS_PACKAGE_OPT_DEPS@
+@FASTRTPS_INSTALLER_DEPS_ANCILLARY@
 
 if(FASTDDS_STATIC)
-    # OpenSSL static libraries are not distributed with the installer
-    @FASTRTPS_PACKAGE_OPT_DEPS@
+    @FASTRTPS_INSTALLER_OPT_DEPS@
     include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-static-targets.cmake)
 else()
     include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-dynamic-targets.cmake OPTIONAL RESULT_VARIABLE DYNAMIC_TARGET_FILE)
```
Comment (richiware) [inline cmake/packaging/Config.cmake.in:35]: When linking against our dynamic libraries, why is not done a `find_package(OpenSSL)` as before?

**Review by richiware — CHANGES_REQUESTED** [review]

File: src/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -456,13 +457,55 @@ elseif(NOT EPROSIMA_INSTALLER)
...
         PRIVATE eProsima_atomic
         )
 
     if(MSVC OR MSVC_IDE)
+
+        # On installed binaries use manifest to specify dependencies
+        if(INSTALLER_PLATFORM AND OPENSSL_FOUND)
+
+            # Get Fast-DDS version suitable manifest format
+            set(FASTDDS_CANONICAL_VERSION ${fastrtps_VERSION})
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply src/cpp/CMakeLists.txt:472]: ok

File: cmake/packaging/Config.cmake.in (hunk 2 above)
Comment (MiguelBarro) [reply cmake/packaging/Config.cmake.in:35]: The OpenSSL dependency is useless for dlls both from sources and binaries. Template config changed accordingly.

File: src/cpp/CMakeLists.txt (hunk 1 above)
Comment (MiguelBarro) [reply src/cpp/CMakeLists.txt:460]: It turns out that [FindOpenSSL module](https://raw.githubusercontent.com/Kitware/CMake/master/Modules/FindOpenSSL.cmake) doesn't include this dependency and must be workaround locally.

**Comment by richiprosima** [comment]
Can't set status; build succeeded.

**Review by richiware — APPROVED** [review]

**Comment by MiguelCompany** [comment]
Failures unrelated to this PR. Going in.

**Comment by leorich** [comment]
I assume this is related (maybe the logic on [line 460](https://github.com/eProsima/Fast-DDS/pull/1971/files/5f96ceaba0e21e2f164905b87d884e53e80becf4#diff-e3052ec6d01ba5f9ae8d014fd3bdd8653f7e759e704a77bbfdb99ada532923daR460) did not work right on my machine) - I get a linker problem when linking to 2.3.3 + on linux (ubuntu 20.04, c++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0) - `c++: error: crypt32.lib: No such file or directory`

Rolling back to 2.3.2 fixes the issue.

**Comment by EduPonz** [comment]
Hi @leorich ,

That's is indeed really strange, since you're using the same OS and g++ version as I am. Could you please provide some step-by-step guide to reproduce (commands, CMake options, etc.).

**Pull request merged**